boost.cmake

```diff
@@ -82,8 +82,37 @@ MACRO(SEARCH_FOR_BOOST)
       Boost_${UPPERCOMPONENT}_LIBRARY_RELEASE
       )
   ENDFOREACH()
+
+  # On darwin systems, we must link againt boost_python with unresolved symbols.
+  # We then remove boost_python from the global Boost_LIBRARIES list to handle it with specific care.
+  LIST(REMOVE_ITEM Boost_LIBRARIES ${Boost_PYTHON_LIBRARY})
```

Comment:
I guess there's no way around this anyway. jrl-cmakemodules users with code linking with Boost Python should be careful when updating (cc @gergondet / @haudren, although AFAIK nobody's concerned in our team).